### PR TITLE
XW-1934 | getAllElements function

### PR DESCRIPTION
**Pull request by dianafa**
https://wikia-inc.atlassian.net/browse/XW-1934

### CODE REVIEW

File: includes/wikia/api/DesignSystemApiController.class.php [hunk 1]
```diff
@@ -150,18 +150,27 @@ public function getFooter() {
 	}
 
 	public function getNavigation() {
-		global $wgUser;
 		$this->checkRequestCompleteness();
 
 		// TODO: change to not mocked data
```
Comment (WikiaTech) [inline includes/wikia/api/DesignSystemApiController.class.php:155]: INFO Complete the task associated to this "TODO" comment. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 9 issues
- MINOR 6 minor
- INFO 3 info

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MINOR [DesignSystemApiController.class.php#L12](https://github.com/Wikia/app/blob/ba186bd266d182d520e21111baba7b68937d5f55/includes/wikia/api/DesignSystemApiController.class.php#L12): Define a constant instead of duplicating this literal "title" 12 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
2. MINOR [DesignSystemApiController.class.php#L13](https://github.com/Wikia/app/blob/ba186bd266d182d520e21111baba7b68937d5f55/includes/wikia/api/DesignSystemApiController.class.php#L13): Define a constant instead of duplicating this literal "translatable-text" 16 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
3. MINOR [DesignSystemApiController.class.php#L33](https://github.com/Wikia/app/blob/ba186bd266d182d520e21111baba7b68937d5f55/includes/wikia/api/DesignSystemApiController.class.php#L33): Define a constant instead of duplicating this literal "link-branded" 5 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
4. MINOR [DesignSystemApiController.class.php#L34](https://github.com/Wikia/app/blob/ba186bd266d182d520e21111baba7b68937d5f55/includes/wikia/api/DesignSystemApiController.class.php#L34): Define a constant instead of duplicating this literal "brand" 6 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
5. MINOR [DesignSystemApiController.class.php#L112](https://github.com/Wikia/app/blob/ba186bd266d182d520e21111baba7b68937d5f55/includes/wikia/api/DesignSystemApiController.class.php#L112): Define a constant instead of duplicating this literal "link-text" 3 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
6. MINOR [DesignSystemApiController.class.php#L146](https://github.com/Wikia/app/blob/ba186bd266d182d520e21111baba7b68937d5f55/includes/wikia/api/DesignSystemApiController.class.php#L146): Define a constant instead of duplicating this literal "wikiId" 4 times. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
7. INFO [DesignSystemApiController.class.php#L66](https://github.com/Wikia/app/blob/ba186bd266d182d520e21111baba7b68937d5f55/includes/wikia/api/DesignSystemApiController.class.php#L66): Move this trailing comment on the previous empty line. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)
8. INFO [DesignSystemApiController.class.php#L92](https://github.com/Wikia/app/blob/ba186bd266d182d520e21111baba7b68937d5f55/includes/wikia/api/DesignSystemApiController.class.php#L92): Move this trailing comment on the previous empty line. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)

**Comment by rybmat** [comment]
LG

**Comment by cormorando** [comment]
LG

**Pull request merged**